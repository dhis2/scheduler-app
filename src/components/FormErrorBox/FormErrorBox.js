import React from 'react'
import { PropTypes } from '@dhis2/prop-types'
import { NoticeBox } from '@dhis2/ui'
import i18n from '@dhis2/d2-i18n'
import styles from './FormErrorBox.module.css'

const FormErrorBox = ({ submitError }) => {
    const hasGenericSubmitErrors = submitError.length > 0

    if (!hasGenericSubmitErrors) {
        return null
    }

    return (
        <NoticeBox
            error
            title={i18n.t('Something went wrong whilst creating your job')}
        >
            <ul className={styles.list}>
                {submitError.map(error => (
                    <li key={error}>{error}</li>
                ))}
            </ul>
        </NoticeBox>
    )
}

const { array } = PropTypes

FormErrorBox.propTypes = {
    submitError: array.isRequired,
}

export default FormErrorBox
