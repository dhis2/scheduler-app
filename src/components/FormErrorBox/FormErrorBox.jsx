import React from 'react'
import PropTypes from 'prop-types'
import { NoticeBox } from '@dhis2/ui'
import styles from './FormErrorBox.module.css'

const FormErrorBox = ({ submitError, title }) => {
    const hasGenericSubmitErrors = submitError.length > 0

    if (!hasGenericSubmitErrors) {
        return null
    }

    return (
        <NoticeBox error title={title}>
            <ul className={styles.list}>
                {submitError.map((error) => (
                    <li key={error}>{error}</li>
                ))}
            </ul>
        </NoticeBox>
    )
}

const { array, string } = PropTypes

FormErrorBox.propTypes = {
    submitError: array.isRequired,
    title: string.isRequired,
}

export default FormErrorBox
