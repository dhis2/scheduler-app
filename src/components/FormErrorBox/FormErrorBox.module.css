.list {
    margin: 0;
    padding-left: 1em;
}
