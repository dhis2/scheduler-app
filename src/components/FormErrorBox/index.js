import FormErrorBox from './FormErrorBox'

export { FormErrorBox }
