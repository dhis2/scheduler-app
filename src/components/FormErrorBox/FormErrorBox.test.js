import React from 'react'
import { shallow } from 'enzyme'
import FormErrorBox from './FormErrorBox'

describe('<FormErrorBox>', () => {
    it('returns null if there are no errors', () => {
        const props = { submitError: [] }

        expect(FormErrorBox(props)).toBeNull()
    })

    it('shows errors if there are errors', () => {
        const submitError = ['Error']
        const wrapper = shallow(<FormErrorBox submitError={submitError} />)

        expect(wrapper).toMatchSnapshot()
    })
})
