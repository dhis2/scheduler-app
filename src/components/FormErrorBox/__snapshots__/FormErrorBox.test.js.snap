// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<FormErrorBox> shows errors if there are errors 1`] = `
<NoticeBox
  dataTest="dhis2-uicore-noticebox"
  error={true}
  title="Something went wrong whilst creating your job"
>
  <ul
    className="list"
  >
    <li
      key="Error"
    >
      Error
    </li>
  </ul>
</NoticeBox>
`;
